forc-plugins/forc-doc/src/lib.rs

```diff
@@ -1,30 +1,94 @@
-pub mod cli;
 pub mod doc;
 pub mod render;
 pub mod search;
-pub mod tests;
 
 use anyhow::{bail, Result};
-use cli::Command;
+use clap::Parser;
 use doc::Documentation;
-use forc_pkg as pkg;
 use forc_pkg::{
+    self as pkg,
     manifest::{GenericManifestFile, ManifestFile},
-    PackageManifestFile,
+    source::IPFSNode,
+    PackageManifestFile, Programs,
 };
 use forc_tracing::println_action_green;
 use forc_util::default_output_directory;
 use render::RenderedDocumentation;
-use std::sync::Arc;
 use std::{
     fs,
     path::{Path, PathBuf},
 };
 use sway_core::{language::ty::TyProgram, BuildTarget, Engines};
 use sway_features::ExperimentalFeatures;
 
+pub const DOC_DIR_NAME: &str = "doc";
 pub const ASSETS_DIR_NAME: &str = "static.files";
 
+forc_util::cli_examples! {
+    crate::Command {
+        [ Build the docs for a project in the current path => "forc doc"]
+        [ Build the docs for a project in the current path and open it in the browser => "forc doc --open" ]
+        [ Build the docs for a project located in another path => "forc doc --path {path}" ]
+        [ Build the docs for the current project exporting private types => "forc doc --document-private-items" ]
+        [ Build the docs offline without downloading any dependencies => "forc doc --offline" ]
+    }
+}
+
+/// Forc plugin for building a Sway package's documentation
+#[derive(Debug, Parser, Default)]
+#[clap(
+    name = "forc-doc",
+    after_help = help(),
+    version
+)]
+pub struct Command {
+    /// Path to the project.
+    ///
+    /// If not specified, current working directory will be used.
+    #[clap(short, long, alias = "manifest-path")]
+    pub path: Option<String>,
+    /// Include non-public items in the documentation.
+    #[clap(long)]
+    pub document_private_items: bool,
+    /// Open the docs in a browser after building them.
+    #[clap(long)]
+    pub open: bool,
+    /// Offline mode, prevents Forc from using the network when managing dependencies.
+    /// Meaning it will only try to use previously downloaded dependencies.
+    #[clap(long = "offline")]
```

Comment:
nit:
```suggestion
    #[clap(long)]
```

Reply:
Good catch. This is all redundant and left over from the initial implementation. Removing now.

Reply:
[50eda9d](https://github.com/FuelLabs/sway/pull/7242/commits/50eda9d3bf8ad081ca6f7374f4d18b5378c8caf0)